### PR TITLE
add slot to sim-lb request

### PR DESCRIPTION
**Pull request by metachris**
## 📝 Summary

Latest sim-lb supports slot header and will print additional timing logs: https://github.com/flashbots/prio-load-balancer/pull/21

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

File: beaconclient/beacon_client_test.go [hunk 1]
```diff
@@ -181,18 +189,19 @@ func TestFetchValidators(t *testing.T) {
...
+		require.Contains(t, validatorResponseEntryToMap(validators.Data), testPubKey)
 
+		// only beacon 2 should have a validator, and should be used by default
 		backend.beaconInstances[0].MockFetchValidatorsErr = nil
 		backend.beaconInstances[1].SetValidators(make(map[types.PubkeyHex]ValidatorResponseEntry))
 		backend.beaconInstances[2].MockFetchValidatorsErr = nil
 		backend.beaconInstances[2].AddValidator(entry)
 
-		t.Log(backend.beaconInstances[1].NumValidators())
+		// t.Log("beacon0/1/2 validators:", backend.beaconInstances[0].NumValidators(), backend.beaconInstances[1].NumValidators(), backend.beaconInstances[2].NumValidators())
```
Comment (avalonche) [inline beaconclient/beacon_client_test.go:201]: leftover debug logs?

File: services/api/service_test.go [hunk 2]
```diff
@@ -203,29 +203,29 @@ func TestStatus(t *testing.T) {
...
-
-		payload := []types.SignedValidatorRegistration{common.ValidPayloadRegisterValidator}
-		rr := backend.request(http.MethodPost, path, payload)
-		require.Equal(t, http.StatusOK, rr.Code)
-		time.Sleep(20 * time.Millisecond) // registrations are processed asynchronously
-
-		isKnown := backend.datastore.IsKnownValidator(pubkeyHex)
-		require.True(t, isKnown)
-	})
+	// t.Run("Normal function", func(t *testing.T) {
```
Comment (avalonche) [inline services/api/service_test.go:206]: will you be replacing the tests with a mock beacon client?

File: beaconclient/multi_beacon_client.go [hunk 3]
```diff
@@ -212,6 +209,17 @@ func (c *MultiBeaconClient) beaconInstancesByLastResponse() []IBeaconInstance {
 	return instances
 }
 
+// beaconInstancesByLastResponse returns a list of beacon clients that has the client
+// with the last successful response as the first element of the slice
+func (c *MultiBeaconClient) beaconInstancesByLeastUsed() []IBeaconInstance {
```
Comment (avalonche) [inline beaconclient/multi_beacon_client.go:214]: should be by last used?

**Review by avalonche — COMMENTED** [review]
wondering if any benchmarking is available for switching over to CL for refreshing validators?

**Comment by metachris** [comment]
this PR was a mess because of merges in between, these changes shouldn't have been part of this at all

> wondering if any benchmarking is available for switching over to CL for refreshing validators?

depends what you want to benchmark. it's not timing critical, and the goal is to reduce the pressure on redis. you can see that clearly happening in https://github.com/flashbots/mev-boost-relay/pull/440

**Comment by metachris** [comment]
rebased this PR

File: services/api/service_test.go (hunk 2 above)
Comment (metachris) [reply services/api/service_test.go]: this was part of another PR, please open an issue or DM me if you want a follow-up

File: beaconclient/multi_beacon_client.go (hunk 3 above)
Comment (metachris) [reply beaconclient/multi_beacon_client.go]: adding more comments in another PR, this is "least" used, which is "last used" in reverse order (so getting all validators is using the least used CL client)

**Pull request closed without merging**